pkg/openstack/cinder.go

```diff
@@ -21,9 +21,27 @@ import (
 
 // ReconcileCinder -
 func ReconcileCinder(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *helper.Helper) (ctrl.Result, error) {
+	cinderName := "cinder"
+	altCinderName := fmt.Sprintf("cinder-%s", instance.UID[:5])
+
+	if instance.Spec.Cinder.RandomPodNames {
+		cinderName, altCinderName = altCinderName, cinderName
+	}
+
+	// Ensure the alternate cinder CR doesn't exist, as the ramdomPodNames flag may have been toggled
 	cinder := &cinderv1.Cinder{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cinder",
+			Name:      altCinderName,
+			Namespace: instance.Namespace,
+		},
+	}
+	if res, err := EnsureDeleted(ctx, helper, cinder); err != nil {
```

Comment:
I assume this will cause a redeploy in case an old `cinder` CR exists, correct?

Reply:
Yes, if the alternative CR exists a redeploy is necessary to have the right naming scheme.